### PR TITLE
Add set_root_weights()

### PR DESCRIPTION
**Pull request by JohnReedV**
## Description
  This PR implements a new function `set_root_weights()` for setting root weights with the coldkey.

### Changes
 * Created `set_root_weights` which contains the original logic associated with the root network from `set_weights`.
 * `set_weights` now rejects calls for the root network. 
 * Root specific operations in `set_weights` have been removed.
 * Added & updated associated unit tests. 

## Related Issue(s)
N/A

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

## Breaking Change

Calls to `set_weights` for the root network will now be rejected. Instead they should call `set_root_weights`. 

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run `cargo fmt` and `cargo clippy` to ensure my code is formatted and linted correctly
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable)

Please include any relevant screenshots or GIFs that demonstrate the changes made.

## Additional Notes

Please provide any additional information or context that may be helpful for reviewers.

### CODE REVIEW

File: pallets/subtensor/src/root.rs [hunk 1]
```diff
@@ -577,6 +577,135 @@ impl<T: Config> Pallet<T> {
...
+        origin: T::RuntimeOrigin,
+        netuid: u16,
+        hotkey: T::AccountId,
+        uids: Vec<u16>,
+        values: Vec<u16>,
+        version_key: u64,
+    ) -> dispatch::DispatchResult {
+        // --- 1. Check the caller's signature. This is the coldkey of a registered account.
+        let coldkey = ensure_signed(origin)?;
+        log::info!(
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/root.rs:589]:
```suggestion
        log::trace!(
```

File: pallets/subtensor/src/lib.rs [hunk 2]
```diff
@@ -1277,6 +1278,83 @@ pub mod pallet {
...
+        // 		- The edge endpoint for the weight, i.e. j for w_ij.
+        //
+        // 	* 'weights' (Vec<u16>):
+        // 		- The u16 integer encoded weights. Interpreted as rational
+        // 		values in the range [0,1]. They must sum to in32::MAX.
+        //
+        // 	* 'version_key' ( u64 ):
+        // 		- The network version key to check if the validator is up to date.
+        //
+        // # Event:
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/lib.rs:1330]: Is this meant to be an error on an event ?

**Review by distributedstatemachine — CHANGES_REQUESTED** [review]

File: pallets/subtensor/src/lib.rs (hunk 2 above)
Comment (JohnReedV) [reply pallets/subtensor/src/lib.rs:1330]: Meant to be an error.

File: pallets/subtensor/src/root.rs (hunk 1 above)
Comment (JohnReedV) [reply pallets/subtensor/src/root.rs:589]: All similar logs use `log::info!` including in the regular `do_set_weights`.

File: pallets/subtensor/src/lib.rs [hunk 3]
```diff
@@ -1305,6 +1307,81 @@ pub mod pallet {
...
+        //  * 'SettingWeightsTooFast':
+        //      - Attempting to set weights too fast.
+        //
+        // 	* 'NotSettingEnoughWeights':
+        // 		- Attempting to set weights with fewer weights than min.
+        //
+        // 	* 'MaxWeightExceeded':
+        // 		- Attempting to set weights with max value exceeding limit.
+        //
+        #[pallet::call_index(8)]
```
Comment (sam0x17) [inline pallets/subtensor/src/lib.rs:1370]: I know we haven't really started doing this yet but ideally these should be doc comments

**Review by sam0x17 — APPROVED** [review]

**Comment by distributedstatemachine** [comment]
TODO:

- [ ] This release has to be coordinated with @opentensor/cortex cc @sam0x17

**Review by distributedstatemachine — APPROVED** [review]

**Pull request merged**